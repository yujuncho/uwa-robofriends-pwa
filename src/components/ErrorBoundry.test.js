import { shallow } from "enzyme";
import ErrorBoundary from "./ErrorBoundry";

describe("ErrorBoundary", () => {
  it("renders children when there is no error", () => {
    const wrapper = shallow(
      <ErrorBoundary>
        <div>Hello!</div>
      </ErrorBoundary>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("sets hasError to true when componentDidCatch", () => {
    const wrapper = shallow(<ErrorBoundary />);
    const instance = wrapper.instance();

    expect(instance.state.hasError).toBe(false);

    instance.componentDidCatch();
    expect(instance.state.hasError).toBe(true);
  });

  it("renders error message when there is an error", () => {
    const wrapper = shallow(
      <ErrorBoundary>
        <div>Hello!</div>
      </ErrorBoundary>
    );
    wrapper.setState({ hasError: true });
    expect(wrapper).toMatchSnapshot();
  });
});
