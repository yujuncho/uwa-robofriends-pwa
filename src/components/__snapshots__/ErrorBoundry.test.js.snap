// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary renders children when there is no error 1`] = `
<div>
  Hello!
</div>
`;

exports[`ErrorBoundary renders error message when there is an error 1`] = `
<h1>
  Something went wrong.
</h1>
`;
